server/container/util-java8/src/main/java/org/apache/james/util/streams/Limit.java

```diff
@@ -35,6 +35,11 @@ public static Limit from(int limit) {
         }
     }
 
+    public static Limit from(Optional<Integer> limit) {
+        return limit.map(Limit::from)
+            .orElse(unlimited());
+    }
+
     public static Limit unlimited() {
         return new Limit(Optional.empty());
```

Comment:
Why don't we use a special Limit value instead of relying on Optional.empty ?

Reply:
Is this related to this pull request?